enhancements/node/cri-o-operator.md

```diff
@@ -0,0 +1,307 @@
+---
+title: cri-o-operator
+authors:
+  - "@saschagrunert"
+reviewers:
+  - TBD
+approvers:
+  - TBD
+creation-date: 2021-06-22
+last-updated: 2021-06-22
+status: provisional
+see-also: {}
+replaces: {}
+superseded-by: {}
+---
+
+# CRI-O operator
+
+## Release Signoff Checklist
+
+- [ ] Enhancement is `implementable`
+- [ ] Design details are appropriately documented from clear requirements
+- [ ] Test plan is defined
+- [ ] Operational readiness criteria is defined
+- [ ] Graduation criteria for dev preview, tech preview, GA
+- [ ] User-facing documentation is created in [openshift-docs](https://github.com/openshift/openshift-docs/)
+
+## Summary
+
+The overall goal of creating this new operator in OpenShift is the encapsulation
+of CRI-O related configurations, metrics and alerts within a single domain and
+namespace. This provides on one hand a higher flexibility for the Node team,
+while on the other hand the dependency on external teams gets reduced. In the
+long-term the Node team would only rely on OpenShift internal APIs to manage the
+container runtime configurations. Additional deployments for proxying runtime
+metrics are part of this operator, too.
+
+## Motivation
+
+Most of the CRI-O related configurations are today part of the [Machine Config
+Operator (MCO)][mco]. Because the MCO got bigger over time, their long-term goal
+is to provide an API for machine related configuration purposes. The new CRI-O
+operator would use this API in a later migration step.
+
+The CRI-O metrics are today served by using a [`kubelet`
+ServiceMonitor][kubelet-monitor] and a corresponding Prometheus relabeling. This
+happens as part of the [Cluster Monitoring Operator (CMO)][cmo] project. The CMO
+team also aims to provide an API for external teams to integrate their
+metrics retrieval into the OpenShift cluster monitoring. In the same way as for
+the MCO, the new CRI-O operator would use this API to integrate the runtime
+metrics retrieval natively into OpenShift by using a dedicated ServiceMonitor
+secured by TLS and RBAC.
+
+[mco]: https://github.com/openshift/machine-config-operator
+[cmo]: https://github.com/openshift/cluster-monitoring-operator
+[kubelet-monitor]: https://github.com/openshift/cluster-monitoring-operator/blob/8ec92e1/assets/control-plane/service-monitor-kubelet.yaml#L95-L116
+
+### Goals
+
+- Create a new Node owned project to be prepared for future changes of the MCO
+  and CMO.
+- Build in CRI-O metrics retrieval by using TLS and RBAC with the help of a
+  kube-rbac-proxy DaemonSet.
+- Include TLS certificate rotation (on disk as well as in a secret) for the
+  CRI-O metrics.
+
+### Non-Goals
+
+Anything which is already part of the MCO or CMO. For example, the CRI-O
```

Comment:
I would more consider this a goal, as ctrcfg handling is in the MCO now. 

also, is kubelet configuration also a goal, or will we have another operator to manage kubelet configuration?

Reply:
I think so, which also means we have to find a better name for the project.

Reply:
the Node Operator (NO) :)

Reply:
That's a great name! Is it a rule to have 3 characters for an operator abbreviation? This would mean we have to call it NOP.

Reply:
or if 4 are allowed, it could also technically be the NOpe

Reply:
We should do a survey.

Reply:
Changed it to "Node Operator" for now.